src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UnsupportedTypeConverterFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Runtime.Serialization;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class UnsupportedTypeConverterFactory : JsonConverterFactory
+    {
+        public override bool CanConvert(Type type)
+        {
+            // If a value type is added, also add a test that
+            // shows NSE is thrown when Nullable<T> is deserialized.
+            return type == typeof(Type) || type == typeof(SerializationInfo);
```

Comment:
What is the problem with these types?  I'd like to see a comment for everything we're explicitly blocking to say why.

Reply:
I've added comments in the source.